docs/design/2022-03-03-rc-read-tso-optimization.md

```diff
@@ -0,0 +1,35 @@
+# Read-Consistency Read With Timestamp Check
+
+- Author(s): cfzjywxk
+- Last updated: March. 3, 2022
+- Discussion at:
+
+## Motivation
+
+For the read-consistency isolation level, each read request in a single transaction will need to fetch a new `ts` to read the latest committed data.
+If the workload is a read-heavy one whose read QPS is significantly higher than writes or there're few read-write conflicts, fetching ts each time will increase the query lantecy.
+
+The new ts itself is used to ensure the most recent data will be returned, if the data version does not change frequently then it's unnecessary to fetch a new timestamp every time.
+The ts fetching could be processed in an optimistic way that ts fetching happens only when a more recent data version is encoutered, this saves the cost of many unncessary ts fetching.
+
+## Detailed Design
+
+Introduce another system variable `tidb_rc_read_check_ts` to enable the "lazy ts fetch" read mode. It will take effect for all the in-transaction select statements when the transaction mode is `pessimistic` and the isolation level is `read-consistency`.
```

Comment:
> the isolation level is `read-consistency`

The behavior is `read-consistency`, but the isolation level is `read-committed`.